src/test/java/co/rsk/bitcoinj/core/TransactionWitnessTest.java

```diff
@@ -193,4 +199,132 @@ public void equals_withTwoTransactionWitnessesWithOneDifferentPush_shouldBeFalse
         assertNotEquals(transactionWitness1, transactionWitness2);
         assertNotEquals(hashCode1, hashCode2);
     }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void getSigInsertionIndex_whenEmptyWitness_shouldThrownArrayIndexOutOfBoundsException() {
+        pushes = new ArrayList<>();
+        TransactionWitness transactionWitness = TransactionWitness.of(pushes);
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+        transactionWitness.getSigInsertionIndex(hashForSignature, signingKey);
+    }
+
+    @Test(expected = ScriptException.class)
+    public void getSigInsertionIndex_whenMalformedRedeemScript_shouldThrowException() {
+        Script customRedeemScript = new Script(new byte[2]);
+        byte[] emptyByte = {};
+        pushes = new ArrayList<>();
+        pushes.add(emptyByte);
+        pushes.add(customRedeemScript.getProgram());
+        TransactionWitness transactionWitness = TransactionWitness.of(pushes);
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        transactionWitness.getSigInsertionIndex(hashForSignature, signingKey);
+    }
+
+    @Test
+    public void getSigInsertionIndex_whenScriptWithOnlyEmptyArrayAndRedeemScript_shouldReturnZero() {
+        pushes = new ArrayList<>();
+        byte[] emptyByte = {};
+        pushes.add(emptyByte); // OP_0
+        pushes.add(emptyByte); // OP_NOTIF
+        pushes.add(redeemScriptHash);
+        TransactionWitness transactionWitness = TransactionWitness.of(pushes);
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
```

Comment:
I think these values can be moved to the top of the class since they are repeated in a couple of tests.

Reply:
done!